File: dsa/6.hash_table/219_Contains_Duplicate_II.java
package hash_table;
import java.util.*;

class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> mapVal = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++){
            if (mapVal.containsKey(nums[i]) == true 
                && Math.abs(i - mapVal.get(nums[i])) <= k){
                    return true;
                }
            else {
                mapVal.put(nums[i], i);
            }
        }
        return false;
    }
}

File: dsa/6.hash_table/219_Contains_Duplicate_II.py
import typing as t

class Solution:
    def containsNearbyDuplicate(self, nums: t.List[int], k: int) -> bool:
        unique_dict: t.Dict[int, int] = {}
        for ind in range(0, len(nums), 1):
            if nums[ind] in unique_dict.keys():
                if abs(ind - unique_dict.get(nums[ind])) <= k:
                    return True
            unique_dict[nums[ind]] = ind
        return False
